### ISSUE TITLE
Cart load not triggered when needed

### ISSUE THREAD

**Issue by mateuszo** [SAP/spartacus#13009]
**Describe the bug**
`ActiveCartService.getEntries()` doesn't trigger cart load and this results in unexpected behaviors. E.g.:
- Cart can be deleted unexpectedly when add to cart is triggered before `ActiveCartService.getActive()`
- `ActiveCartService.getEntries()` shows inconsistent data when called before `ActiveCartService.getActive()`

**Tell us the version of Spartacus**
- Library version: 3.4

**To Reproduce**
1. Remove mini cart from the page (e.g. by removing the MiniCart slot).
2. Go to PDP or PLP and add a product to the cart.
3. Go to the cart page and check the cart contents and remember them.
4. Go to PDP or PLP again and refresh the page.
5. Add a different product to the cart than previously.
6. Go to the cart page and observe that the previous cart is gone.

Another way to observe this:
1. Remove mini cart from the page (e.g. by removing the MiniCart slot).
2. Go to PDP or PLP and add a product to the cart.
3. Open devtools and check active cart id in local storage.
4. Refresh the page and add a product again to the cart.
5. Observe that the cart id has changed.

**Expected behavior**
Whenever any cart part is being requested the cart load should be triggered. It's not only about `getEntries()`, I presume that e.g. `getEntry`, `getLastEntry` and `addEmail` might cause the same behavior.

**Additional context**
This came out by accident when a developer added custom implementation for the mini-cart which only used `getEntries()` method instead. This is a possible real-life scenario and this is not intuitive that `getActive()` has to be called first.

**Comment by giancorderoortiz** [comment]
@WeizhengSap, @plabadie please double check. Is it a bug or as designed?

**Comment by Xymmer** [comment]
let's investigate and if we need to, backport to 3.4 after fixing in 4.0.x or 4.x

**Comment by Platonn** [comment]
Good point @mateuszo , now the load of cart is triggered only when some component or service subscribes to the state of the cart (`getActive()`). Becasue mini-cart in OOTB Spartacus is presented in every page and calls getActive(), there is no problem in OOTB Spartacus.

I admit it might be counter intuitive that `getEntries` and `getLastEntry` don't work when you don't subscribe to `getActive()` first. I agree we can improve it.
Though, when it comes to implementation of the improvement, we should make sure that we don't load the cart more often than we have to. 

btw. I'm not convinced yet about `addEmail`, it seems a bit different case to me. We could assume that we have the cart loaded and created when we add email, becasue adding email is just a part of the guest checkout. What do you think? 

Moreover, before we decide whether to load the cart as part of methods `getEntries()`, `getEntry`, `getLastEntry`, we should first investigate side-effect consequences of it.

--
IMHO this issue is not a `critical`, nor a functional `bug`. It's just a current API design of the of ActiveCartService, which could be potentially more intuitive for developers. And many thanks @mateuszo for pointing that out!

I'm in favor of treating the improvement as a feature, but not a bugfix (especially I would not backport it, as it would be a functional breaking change: to load cart in new moments,). Moreover it needs a careful eye when implementing -  not to load cart too often.

Customer can workaround, by subscribing to `ActiveCartService.getActive()`, anyway.
